Can't build if VS2019 is installed

**Describe the bug**
<!-- Please enter a short, clear description of the bug -->
If you have VS2019 installed, the Build.cmd script will run .\tools\NugetWrapper.cmd, which will call nuget.exe with no specific MSBuild version specified, which will find VS2019's MSBuild 16, which won't be able to restore the dependencies, failing the build.

**Steps to reproduce the bug**
1) Install VS2017 (15.9.11)
2) Install VS2019 (16.x)
3) Clone the repo and run .\Build.cmd

**Expected behavior**
Project builds fine with VS2019 SxS with VS2017.

```
MSBuild auto-detection: using msbuild version '16.1.27.44945' from 'C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Current\bin'.
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Microsoft\VC\v150\Microsoft.CppCommon.targets(2128,3): error MSB4019: The imported project "C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Microsoft\WindowsXaml\v15.0\Microsoft.Windows.UI.Xaml.Cpp.targets" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk. [C:\Users\alzollin\source\repos\microsoft-ui-xaml\dev\dll\Microsoft.UI.Xaml.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Microsoft\VC\v150\Microsoft.CppCommon.targets(2128,3): error MSB4019: The imported project "C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Microsoft\WindowsXaml\v15.0\Microsoft.Windows.UI.Xaml.Cpp.targets" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk. [C:\Users\alzollin\source\repos\microsoft-ui-xaml\test\MUXTestUtilities\MUXTestUtilities.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Microsoft\VC\v150\Microsoft.CppCommon.targets(2128,3): error MSB4019: The imported project "C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Microsoft\WindowsXaml\v15.0\Microsoft.Windows.UI.Xaml.Cpp.targets" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk. [C:\Users\alzollin\source\repos\microsoft-ui-xaml\test\TestAppCX\TestAppCX.vcxproj]
C:\Users\alzollin\source\repos\microsoft-ui-xaml\test\MUXControlsAdhocApp\MUXControlsAdhocApp.csproj(186,11): error MSB4226: The imported project "C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Microsoft\WindowsXaml\v15.0\Microsoft.Windows.UI.Xaml.CSharp.targets" was not found. Also, tried to find "Microsoft\WindowsXaml\v15.0\Microsoft.Windows.UI.Xaml.CSharp.targets" in the fallback search path(s) for $(MSBuildExtensionsPath) - "C:\Program Files (x86)\MSBuild" . These search paths are defined in "C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Current\Bin\MSBuild.exe.Config". Confirm that the path in the <Import> declaration is correct, and that the file exists on disk in one of the search paths.
C:\Users\alzollin\source\repos\microsoft-ui-xaml\test\MUXControlsTestAppForIslands\MUXControlsTestAppForIslands.csproj(171,3): error MSB4019: The imported project "C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Microsoft\WindowsXaml\v15.0\Microsoft.Windows.UI.Xaml.CSharp.targets" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.
C:\Users\alzollin\source\repos\microsoft-ui-xaml\test\MUXControlsTestApp\MUXControlsTestApp.Shared.projitems(296,3): error MSB4019: The imported project "C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Microsoft\WindowsXaml\v15.0\Microsoft.Windows.UI.Xaml.CSharp.targets" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk. [C:\Users\alzollin\source\repos\microsoft-ui-xaml\test\MUXControlsTestApp\MSTest\MUXControlsTestApp.csproj]
C:\Users\alzollin\source\repos\microsoft-ui-xaml\test\MUXControlsTestApp\MUXControlsTestApp.Shared.projitems(296,3): error MSB4019: The imported project "C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Microsoft\WindowsXaml\v15.0\Microsoft.Windows.UI.Xaml.CSharp.targets" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk. [C:\Users\alzollin\source\repos\microsoft-ui-xaml\test\MUXControlsTestApp\TAEF\MUXControlsTestApp.TAEF.csproj]
C:\Users\alzollin\source\repos\microsoft-ui-xaml\test\IXMPTestApp\IXMPTestApp.Shared.projitems(201,3): error MSB4019: The imported project "C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Microsoft\WindowsXaml\v15.0\Microsoft.Windows.UI.Xaml.CSharp.targets" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk. [C:\Users\alzollin\source\repos\microsoft-ui-xaml\test\IXMPTestApp\MSTest\IXMPTestApp.csproj]
C:\Users\alzollin\source\repos\microsoft-ui-xaml\test\IXMPTestApp\IXMPTestApp.Shared.projitems(201,3): error MSB4019: The imported project "C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Microsoft\WindowsXaml\v15.0\Microsoft.Windows.UI.Xaml.CSharp.targets" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk. [C:\Users\alzollin\source\repos\microsoft-ui-xaml\test\IXMPTestApp\TAEF\IXMPTestApp.TAEF.csproj]
```

--- a/Build.cmd
+++ b/Build.cmd
@@ -96,6 +96,11 @@ if "%1" == "/project" (
     shift
     goto :MoreArguments
 )
+if "%1" == "/usevsprerelease" (
+    set VSWHEREPARAMS=%VSWHEREPARAMS% -prerelease
+    shift
+    goto :MoreArguments
+)
 if "%1" NEQ "" (
     echo ERROR: Unknown argument "%1"
     goto :usage
@@ -104,12 +109,12 @@ goto :DoBuild
 
 :DoBuild
 
-"%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere.exe" -Latest -requires Microsoft.Component.MSBuild -find MSBuild\**\Bin\MSBuild.exe > %TEMP%\msbuildpath.txt
-
-set /p MSBUILDPATH=<%TEMP%\msbuildpath.txt
-
-For %%A in ("%MSBUILDPATH%") do (set MSBUILDDIRPATH=%%~dpA)
-IF %MSBUILDDIRPATH:~-1%==\ SET MSBUILDDIRPATH=%MSBUILDDIRPATH:~0,-1%
+for /f "usebackq tokens=*" %%i in (`"%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere.exe" -latest %VSWHEREPARAMS% -requires Microsoft.Component.MSBuild -find MSBuild\**\Bin\MSBuild.exe`) do (
+  set MSBUILDPATH=%%i
+  set MSBUILDDIRPATH=%%~dpi
+  IF !MSBUILDDIRPATH:~-1!==\ SET MSBUILDDIRPATH=!MSBUILDDIRPATH:~0,-1!
+  echo Using MSBuild from !MSBUILDPATH! in !MSBUILDDIRPATH!
+)
 
 REM
 REM     NUGET Restore
@@ -133,18 +138,18 @@ if "%USEINTERNALSDK%" == "1" ( set EXTRAMSBUILDPARAMS=/p:UseInternalSDK=true )
 if "%EMITTELEMETRYEVENTS%" == "1" ( set EXTRAMSBUILDPARAMS=/p:EmitTelemetryEvents=true )
 
 if "%PROJECTPATH%" NEQ "" (
-    set MSBuildCommand="%MSBUILDPATH%" %PROJECTPATH% /p:platform="%BUILDPLATFORM%" /p:configuration="%BUILDCONFIGURATION%" /p:VisualStudioVersion="15.0" /flp:Verbosity=Diagnostic /fl /bl %EXTRAMSBUILDPARAMS% /verbosity:Minimal
+    set MSBuildCommand="%MSBUILDPATH%" %PROJECTPATH% /p:platform="%BUILDPLATFORM%" /p:configuration="%BUILDCONFIGURATION%" /flp:Verbosity=Diagnostic /fl /bl %EXTRAMSBUILDPARAMS% /verbosity:Minimal
     echo !MSBuildCommand!
     !MSBuildCommand!
 ) else (
     if "%BUILDALL%" == "" (
         set XES_OUTDIR=%BUILD_BINARIESDIRECTORY%\%BUILDCONFIGURATION%\%BUILDPLATFORM%\
 
-        "%MSBUILDPATH%" .\MUXControls.sln /p:platform="%BUILDPLATFORM%" /p:configuration="%BUILDCONFIGURATION%" /p:VisualStudioVersion="15.0" /flp:Verbosity=Diagnostic /fl /bl %EXTRAMSBUILDPARAMS% /verbosity:Minimal
+        "%MSBUILDPATH%" .\MUXControls.sln /p:platform="%BUILDPLATFORM%" /p:configuration="%BUILDCONFIGURATION%" /flp:Verbosity=Diagnostic /fl /bl %EXTRAMSBUILDPARAMS% /verbosity:Minimal
 
         if "%ERRORLEVEL%" == "0" call .\tools\MakeAppxHelper.cmd %BUILDPLATFORM% %BUILDCONFIGURATION%
     ) else (
-        "%MSBUILDPATH%" .\build\BuildAll.proj /maxcpucount:12 /p:VisualStudioVersion="15.0" /flp:Verbosity=Diagnostic /fl /bl /verbosity:Minimal
+        "%MSBUILDPATH%" .\build\BuildAll.proj /maxcpucount:12 /flp:Verbosity=Diagnostic /fl /bl /verbosity:Minimal
 
         if "%ERRORLEVEL%" == "0" (
             call .\tools\MakeAppxHelper.cmd x86 release
@@ -186,6 +191,7 @@ echo        /UseInsiderSDK - build using insider SDK
 echo        /UseInternalSDK - build using internal SDK
 echo        /EmitTelemetryEvents - build with telemetry events turned on
 echo        /project ^<path^> - builds a specific project
+echo        /usevsprerelease - use the prerelease VS on the machine instead of latest stable
 echo.
 
 :end

--- a/Build.cmd
+++ b/Build.cmd
@@ -104,13 +104,20 @@ goto :DoBuild
 
 :DoBuild
 
+"%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere.exe" -Latest -requires Microsoft.Component.MSBuild -find MSBuild\**\Bin\MSBuild.exe > %TEMP%\msbuildpath.txt
+
+set /p MSBUILDPATH=<%TEMP%\msbuildpath.txt
+
+For %%A in ("%MSBUILDPATH%") do (set MSBUILDDIRPATH=%%~dpA)
+IF %MSBUILDDIRPATH:~-1%==\ SET MSBUILDDIRPATH=%MSBUILDDIRPATH:~0,-1%
+
 REM
 REM     NUGET Restore
 REM
 if "%PROJECTPATH%" NEQ "" (
-	call .\Tools\NugetWrapper.cmd restore -MSBuildPath "%VSINSTALLDIR%\MSBuild\15.0\Bin" -NonInteractive %PROJECTPATH%
+	call .\Tools\NugetWrapper.cmd restore -MSBuildPath "%MSBUILDDIRPATH%" -NonInteractive %PROJECTPATH%
 ) else (
-	call .\Tools\NugetWrapper.cmd restore -MSBuildPath "%VSINSTALLDIR%\MSBuild\15.0\Bin" -NonInteractive .\MUXControls.sln 
+	call .\Tools\NugetWrapper.cmd restore -MSBuildPath "%MSBUILDDIRPATH%" -NonInteractive .\MUXControls.sln 
 )
 
 REM
@@ -124,9 +131,6 @@ if "%MUXFINAL%" == "1" ( set EXTRAMSBUILDPARAMS=/p:MUXFinalRelease=true )
 if "%USEINSIDERSDK%" == "1" ( set EXTRAMSBUILDPARAMS=/p:UseInsiderSDK=true )
 if "%USEINTERNALSDK%" == "1" ( set EXTRAMSBUILDPARAMS=/p:UseInternalSDK=true )
 if "%EMITTELEMETRYEVENTS%" == "1" ( set EXTRAMSBUILDPARAMS=/p:EmitTelemetryEvents=true )
-
-REM Need an explicit full path to MSBuild.exe or it will fall back to 14.0 for some reason
-set MSBUILDPATH=%VSINSTALLDIR%\MSBuild\15.0\Bin\MSBuild.exe
 
 if "%PROJECTPATH%" NEQ "" (
     set MSBuildCommand="%MSBUILDPATH%" %PROJECTPATH% /p:platform="%BUILDPLATFORM%" /p:configuration="%BUILDCONFIGURATION%" /p:VisualStudioVersion="15.0" /flp:Verbosity=Diagnostic /fl /bl %EXTRAMSBUILDPARAMS% /verbosity:Minimal

--- a/Build.cmd
+++ b/Build.cmd
@@ -108,9 +108,9 @@ REM
 REM     NUGET Restore
 REM
 if "%PROJECTPATH%" NEQ "" (
-    call .\Tools\NugetWrapper.cmd restore -NonInteractive %PROJECTPATH%
+	call .\Tools\NugetWrapper.cmd restore -MSBuildPath "%VSINSTALLDIR%\MSBuild\15.0\Bin" -NonInteractive %PROJECTPATH%
 ) else (
-    call .\Tools\NugetWrapper.cmd restore -NonInteractive .\MUXControls.sln
+	call .\Tools\NugetWrapper.cmd restore -MSBuildPath "%VSINSTALLDIR%\MSBuild\15.0\Bin" -NonInteractive .\MUXControls.sln 
 )
 
 REM

